sale_order_line_date/models/sale_order_line.py

```diff
@@ -0,0 +1,43 @@
+# © 2016 OdooMRP team
+# © 2016 AvanzOSC
+# © 2016 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# © 2016 ForgeFlow S.L. (https://forgeflow.com)
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from datetime import timedelta
+
+from odoo import fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    commitment_date = fields.Datetime("Delivery Date")
+
+    def _prepare_procurement_values(self, group_id=False):
+        vals = super()._prepare_procurement_values(group_id)
+        # has ensure_one already
+        if self.commitment_date:
+            vals.update(
+                {
+                    "date_planned": self.commitment_date
+                    - timedelta(days=self.order_id.company_id.security_lead),
+                    "date_deadline": self.commitment_date,
+                }
+            )
+        return vals
+
+    def write(self, vals):
+        res = super().write(vals)
+        moves_to_upd = set()
+        if "commitment_date" in vals:
+            for move in self.move_ids:
+                if move.state not in ["cancel", "done"]:
+                    moves_to_upd.add(move.id)
```

Comment:
`write` work over multi records, please avoid singleton error if write is done with multi records
```suggestion
            for line in self:
              for move in line.move_ids:
                  if move.state not in ["cancel", "done"]:
                      moves_to_upd.add(move.id)
```

Reply:
Both self and move_ids are recordsets, so I think this code is viable as self.move_ids will just get all move_ids of all records in self. @celm1990

Reply:
Was wondering, why iterate when you can map/filter instead?

```python
        if 'commitment_date' in values and (moves_to_update := self.mapped('move_ids').filtered(lambda move: move.state not in ['cancel', 'done'])):
            moves_to_update.date_deadline = values.get('commitment_date')
```

Reply:
@BialArno In fact, mapped.filtered do the same as the current code. I was using Odoo provided methods before but now I'd prefer this writing as clearer.

Reply:
@rousseldenis well yes, they do the same, more or less it's pick of personal preferrence, but then we should consider implementing framework methods over sticking to python basics or go plain python and at that point, in my opinion, in Odoo these methods weren't added for nothing :) Ofc competition of speed might be useful to check if for loop or mapped/filtered does same task faster and/or more resource efficient.